### PR TITLE
Issue #13717: EE 9 EJB Bindings FAT

### PR DESCRIPTION
**Pull request by tkburroughs**
Enable the EJB bindings FAT tests to run with Jakarta EE 9 features.

for #13717

### CODE REVIEW

**Comment by tkburroughs** [comment]
#build

**Comment by tkburroughs** [comment]
The EE9 Repeat action permanently changes the server.xml, so all subsequent tests that don't use a feature replace action will fail because the apps won't be transformed, but the server.xml will still be using jakarta.  This is only a problem for the 2 tests that use RepeatOnError actions, as all the rest will switch to a specific Java EE/Jakarta EE level.

Changing RepeatOnError to subclass a specific Java EE feature replace action should resolve the issue.

**Comment by tkburroughs** [comment]
#run_libby_bot

**Comment by tkburroughs** [comment]
#run-libby-bot

**Comment by tkburroughs** [comment]
RepeatOnError repeat action has been replaced with RepeatOnErrorEE7 and RepeatOnErrorEE8; tests have been updated and will always specifically call out EE version, thus avoiding the problem of using a modified server from a prior test.

**Comment by tkburroughs** [comment]
#build

File: dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE7.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            case WARN:
+                return RepeatTestFilter.isRepeatActionActive(ID_WARN);
+        }
+        return false;
+    }
+
+    @Override
+    public FeatureReplacementAction forServers(String... serverNames) {
+        if (serverNames == null || serverNames.length != 1) {
+            throw new IllegalStateException("Only 1 server supported by RepeatOnErrorEE9 repeat action");
```
Comment (olendvcook) [inline dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE7.java]: the exception has RepeatOnErrorEE9 printout in the RepeatOnErrorEE7 class

File: dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE8.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            case WARN:
+                return RepeatTestFilter.isRepeatActionActive(ID_WARN);
+        }
+        return false;
+    }
+
+    @Override
+    public FeatureReplacementAction forServers(String... serverNames) {
+        if (serverNames == null || serverNames.length != 1) {
+            throw new IllegalStateException("Only 1 server supported by RepeatOnErrorEE9 repeat action");
```
Comment (olendvcook) [inline dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE8.java]: the exception has RepeatOnErrorEE9 printout in the RepeatOnErrorEE8 class

**Review by olendvcook — APPROVED** [review]

File: dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE7.java (hunk 1 above)
Comment (tkburroughs) [reply dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE7.java]: corrected.

File: dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE8.java (hunk 2 above)
Comment (tkburroughs) [reply dev/com.ibm.ws.ejbcontainer.bindings_fat/fat/src/com/ibm/ws/ejbcontainer/bindings/fat/tests/repeataction/RepeatOnErrorEE8.java]: corrected

**Comment by tkburroughs** [comment]
minor issues corrected
#run-libby-bot
#build

**Pull request merged**